Assets/CustomPackages/AI/StateMachine/AIBrain.cs

```diff
@@ -176,18 +159,33 @@ public override void BaseInView(Transform basePoint)
 
     public override void PlayerInView()
     {
-        if (_currentTarget != GameManager.playerRef.transform)
-            Destroy(_currentTarget.gameObject);
-        _currentTarget = GameManager.playerRef.transform;
-        _attackState.SetTarget(GameManager.playerRef);
-        _followTargetState.SetTarget(GameManager.playerRef);
+
+        if (_currentTarget != null)
+        {
+            _attackState.SetTarget(GameManager.playerRef);
+            _followTargetState.SetTarget(GameManager.playerRef);
+            if (_currentTarget != GameManager.playerRef.transform)
+            {
+                Transform mock = _currentTarget;
```

Comment:
why do you create a mock Transfor, do nothing with it and then destroy it?

Reply:
I had some weird errors and I made some changes here and there. But yeah probably this thing does not help at all, soo good point I will change it